src/test/java/com/socrata/BulkUploadTest.java

```diff
@@ -159,41 +160,46 @@ public void testUpsertWithRowIdentifier() throws IOException, SodaError, Interru
         importer.publish(dataset.getId());
 
         try {
-
-            //
-            //Verify the row we expect is really there.
-            final SoqlQuery   lookupTestRow = new SoqlQueryBuilder()
+            // Verify the row we expect is really there.
+            final SoqlQuery lookupTestRow = new SoqlQueryBuilder()
                         .setWhereClause("id='8880962'")
                         .build();
+            /* There is a delay between writing and reading the dataset so we should wait up to 5 seconds.
+             * If this is consistently failing, we may want to increase the sleep or find a new way of
+             * testing that the row count has been updated. */
+            Thread.sleep(5000);
```

Comment:
Is this due to replication? If so maybe we want a ticket that's like "write a helper for polling until replication is complete, and use that instead of sleep"

Reply:
It probably is! I made [this ticket](https://socrata.atlassian.net/browse/EN-45880) to capture that work.

Reply:
Because this project is public facing and the thing management wants to add isn't really an emergency or making a huge benefit, I feel that we should do more to cover those small tedious tasks that are not thorny.

Reply:
Turning errors to ignores isn't good

Reply:
Talked about this offline--work to revitalize tests is captured in [this ticket](https://socrata.atlassian.net/browse/EN-45878).